src/epcpm/sunspectointerface.py

```diff
@@ -0,0 +1,2227 @@
+                sunspec_id=self.sunspec_id,
+            )
+            get_child, set_child, base_decl_child = builder.gen()
+            get_out.append(get_child)
+            set_out.append(set_child)
+            base_decl.append(base_decl_child)
+
+        return [get_out, set_out, base_decl]
+
+
+@specific_builders(epcpm.sunspecmodel.HeaderBlock)
+@attr.s
+class SpecificHeaderBlock:
+    """Specific model interface generator for the SunSpec HeaderBlock class."""
+
+    wrapped = attr.ib()
+    model_id = attr.ib()
+    repeating_block_reference = attr.ib(default=None)
+    scale_factor_from_uuid = attr.ib(default=None)
+    parameter_uuid_finder = attr.ib(default=None)
+    sunspec_id = attr.ib(default=None)
+    fixed_block_reference = attr.ib(default=None, type=epcpm.sunspecmodel.FixedBlock)
+
+    def gen(self) -> typing.Tuple[typing.List[str], typing.List[str], typing.List[str]]:
+        """
+        Specific model interface generator for the SunSpec HeaderBlock class.
+
+        Returns:
+            [list[str], list[str], list[str]]: list of getters, list of setters, list of base declaration strings
+        """
+        return [[], [], []]
+
+    def gen_scale_factor(self):
+        return {}
+
+
+@specific_builders(epcpm.sunspecmodel.TableRepeatingBlockReference)
+@attr.s
+class SpecificTableRepeatingBlockReference:
+    """Specific model interface generator for the SunSpec TableRepeatingBlockReference class."""
+
+    wrapped = attr.ib()
+    model_id = attr.ib()
+    scale_factor_from_uuid = attr.ib(default=None)
+    parameter_uuid_finder = attr.ib(default=None)
+    sunspec_id = attr.ib(default=None)
+    is_table = attr.ib(default=False)
+    fixed_block_reference = attr.ib(default=None, type=epcpm.sunspecmodel.FixedBlock)
+
+    def gen(self) -> typing.Tuple[typing.List[str], typing.List[str], typing.List[str]]:
+        """
+        Specific model interface generator for the SunSpec TableRepeatingBlockReference class.
+
+        Returns:
+            [list[str], list[str], list[str]]: list of getters, list of setters, list of base declaration strings
+        """
+        builder = specific_builders.wrap(
+            wrapped=self.wrapped.original,
+            parameter_uuid_finder=self.parameter_uuid_finder,
+            model_id=self.model_id,
+            is_table=self.is_table,
+            repeating_block_reference=self.wrapped,
+            sunspec_id=self.sunspec_id,
+        )
+
+        return builder.gen()
+
+    def gen_scale_factor(self):
+        return {}
+
+
+@specific_builders(epcpm.sunspecmodel.DataPoint)
+@attr.s
+class SpecificDataPoint:
+    """Specific model interface generator for the SunSpec DataPoint class."""
+
+    wrapped = attr.ib()
+    scale_factor_from_uuid = attr.ib()
+    model_id = attr.ib()
+    is_table = attr.ib()
+    parameter_uuid_finder = attr.ib(default=None)
+    sunspec_id = attr.ib(default=None)
+
+    def gen(self) -> typing.Tuple[typing.List[str], typing.List[str], str]:
```

Comment:
This looks to be another function that could do with some splitting out into multiple functions.

Reply:
Refactored in 55b86fd099930810fa0d9d09e15bb6c86531de65, although probably not as much as you'd like. It took a long time just to do this one.